packages/common-scripts/src/sudt.ts

```diff
@@ -0,0 +1,421 @@
+import { addCellDep } from "./helper";
+import {
+  utils,
+  Hash,
+  Address,
+  Cell,
+  Script,
+  Header,
+  CellCollector,
+} from "@ckb-lumos/base";
+const { toBigUInt128LE, readBigUInt64LE, computeScriptHash } = utils;
+import {
+  serializeMultisigScript,
+  multisigArgs,
+  FromInfo,
+} from "./secp256k1_blake160_multisig";
+import common from "./common";
+import {
+  parseAddress,
+  minimalCellCapacity,
+  TransactionSkeletonType,
+  Options,
+} from "@ckb-lumos/helpers";
+import { Set, List } from "immutable";
+import { getConfig, Config } from "@ckb-lumos/config-manager";
+import { collectCells, LocktimeCell } from "./locktime_pool";
+
+export async function createToken(
+  txSkeleton: TransactionSkeletonType,
+  fromInfo: FromInfo,
+  amount: bigint,
+  capacity?: bigint,
+  tipHeader?: Header,
+  { config = undefined }: Options = {}
+): Promise<TransactionSkeletonType> {
+  config = config || getConfig();
+
+  const template = config.SCRIPTS.SUDT;
+
+  if (!template) {
+    throw new Error("Provided config does not have SUDT script setup!");
+  }
+
+  txSkeleton = addCellDep(txSkeleton, {
+    out_point: {
+      tx_hash: template.TX_HASH,
+      index: template.INDEX,
+    },
+    dep_type: template.DEP_TYPE,
+  });
+
+  const fromScript = _fromInfoToScript(fromInfo, config);
+
+  const toScript = fromScript;
+
+  const sudtTypeScript = {
+    code_hash: template.CODE_HASH,
+    hash_type: template.HASH_TYPE,
+    args: computeScriptHash(fromScript),
+  };
+
+  const targetOutput: Cell = {
+    cell_output: {
+      capacity: "0x0",
+      lock: toScript,
+      type: sudtTypeScript,
+    },
+    data: toBigUInt128LE(amount),
+    out_point: undefined,
+    block_hash: undefined,
+  };
+
+  if (!capacity) {
+    capacity = minimalCellCapacity(targetOutput);
+  }
+  capacity = BigInt(capacity);
+  targetOutput.cell_output.capacity = "0x" + capacity.toString(16);
+
+  txSkeleton = txSkeleton.update("outputs", (outputs) => {
+    return outputs.push(targetOutput);
+  });
+
+  const outputIndex = txSkeleton.get("outputs").size - 1;
+
+  // fix entry
+  txSkeleton = txSkeleton.update("fixedEntries", (fixedEntries) => {
+    return fixedEntries.push({
+      field: "outputs",
+      index: outputIndex,
+    });
+  });
+
+  txSkeleton = await common.injectCapacity(
+    txSkeleton,
+    outputIndex,
+    [fromInfo],
+    tipHeader,
+    {
+      config,
+    }
+  );
+
+  return txSkeleton;
+}
+
+/**
+ *
+ * @param txSkeleton
+ * @param fromInfos
+ * @param sudtToken
+ * @param toAddress
+ * @param amount
+ * @param changeAddress if not provided, will use first fromInfo
+ * @param capacity
+ * @param options
+ */
+export async function transfer(
+  txSkeleton: TransactionSkeletonType,
+  fromInfos: FromInfo[],
+  sudtToken: Hash,
```

Comment:
Let's define a `Token` type, for now it could be just the args, but later we can expand this to full Script

Reply:
updated.